firebase_admin/remote_config.py

```diff
@@ -0,0 +1,597 @@
+    """
+    Represents an OR condition.
+    """
+    def __init__(self, conditions: List[Condition]):
+        super().__init__()
+        self.conditions = conditions
+
+
+class AndCondition(Condition):
+    """
+    Represents an AND condition.
+    """
+    def __init__(self, conditions: List[Condition]):
+        super().__init__()
+        self.conditions = conditions
+
+
+class PercentCondition(Condition):
+    """
+    Represents a percent condition.
+    """
+    def __init__(self, seed: str, percent_operator: PercentConditionOperator, micro_percent: int, micro_percent_range: Optional[Dict[str, int]] = None):
+        super().__init__()
+        self.seed = seed
+        self.percent_operator = percent_operator
+        self.micro_percent = micro_percent
+        self.micro_percent_range = micro_percent_range
+
+
+class CustomSignalCondition(Condition):
+    """
+    Represents a custom signal condition.
+    """
+    def __init__(self, custom_signal_operator: CustomSignalOperator, custom_signal_key: str, target_custom_signal_values: List[Union[str, int, float]]):
+        super().__init__()
+        self.custom_signal_operator = custom_signal_operator
+        self.custom_signal_key = custom_signal_key
+        self.target_custom_signal_values = target_custom_signal_values
+
+
+class OneOfCondition(Condition):
+    """
+    Represents a condition that can be one of several types.
+    """
+    def __init__(self, or_condition: Optional[OrCondition] = None, and_condition: Optional[AndCondition] = None, true_condition: Optional[bool] = None, false_condition: Optional[bool] = None, percent_condition: Optional[PercentCondition] = None, custom_signal_condition: Optional[CustomSignalCondition] = None):
+        super().__init__()
+        self.or_condition = or_condition
+        self.and_condition = and_condition
+        self.true_condition = true_condition
+        self.false_condition = false_condition
+        self.percent_condition = percent_condition
+        self.custom_signal_condition = custom_signal_condition
+
+
+class NamedCondition:
+    """
+    Represents a named condition.
+    """
+    def __init__(self, name: str, condition: OneOfCondition):
+        self.name = name
+        self.condition = condition
+
+
+class EvaluationContext:
+    """
+    Represents the context for evaluating conditions.
+    """
+    def __init__(self, **kwargs):
+        # This allows you to pass any key-value pairs to the context
+        # For example: EvaluationContext(user_country="US", user_type="paid")
+        self.__dict__.update(kwargs)
+
+    def __getattr__(self, item):
+        # This handles the case where a key is not found in the context
+        return None
+
+class ConditionEvaluator:
+    """
+    Encapsulates condition evaluation logic to simplify organization and
+    facilitate testing.
+    """
+
+    def evaluate_conditions(self, named_conditions: List['NamedCondition'], context: 'EvaluationContext') -> Dict[str, bool]:
```

Comment:
initially planned to use a List<NamedCondition> for notifications, but since the NamedCondition class has been removed, this code is no longer needed. It has been removed in recent commits